file/fat32.lisp

```diff
@@ -1281,25 +1279,85 @@ Valid media-type ara 'FAT32   ' " fat-type-label)))
         (write-char #\~ s))
       s)))
 
-(defclass fat-file-stream (mezzano.gray:fundamental-binary-input-stream
-                           mezzano.gray:fundamental-binary-output-stream
-                           file-cache-stream
-                           file-stream)
+(defclass fat-file-stream (file-cache-stream
+                           mezzano.gray:fundamental-binary-input-stream
+                           mezzano.gray:fundamental-binary-output-stream)
   ((pathname :initarg :pathname :reader file-stream-pathname)
    (host :initarg :host :reader host)
+   (if-exists :initarg :if-exists :reader if-exists)
    ;; File position where the buffer data starts.
-   (buffer-position :initarg :buffer-position
-                    :initform 0
-                    :accessor buffer-position)
+   (buffer-position :initarg :buffer-position :accessor buffer-position)
```

Comment:
What does this slot actually store? It looks like it's the first cluster of the file. Call it `first-cluster` if so?